src/loading2/octree-geometry-node.ts

```diff
@@ -67,15 +68,18 @@ export class OctreeGeometryNode implements IPointCloudTreeNode {
 		return this.boundingBox;
 	}
 
-	load() {
-
+	load(): Promise<void> {
 		if (this.octreeGeometry.numNodesLoading >= this.octreeGeometry.maxNumNodesLoading) {
-			return;
+			return Promise.resolve();
 		}
 
-		if (this.octreeGeometry.loader) {
-			this.octreeGeometry.loader.load(this);
+		if (!this.octreeGeometry.loader) {
+			this.loading = false;
+			this.failed = true;
+			return Promise.reject(`Loader not initialized for ${this.name}`);
 		}
+
+		return this.octreeGeometry.loader.load(this);
```

Comment:
these changes seem irrelevant to the proposed changes in the PR.

Reply:
I implemented these changes as the new IPointCloudGeometryNode interface enforces the `load` function returning a `Promise` type. 

I wanted to make sure that OctreeGeometryNode also properly set the loading and failed attributes properly, however I can move this to separate PR if you would prefer!

Reply:
It's just fine within the contents of this PR.